packages/react-component-library/src/components/Drawer/Drawer.tsx

```diff
@@ -20,7 +20,10 @@ export const Drawer: React.FC<DrawerProps> = ({
 
   function handleClick(event: React.FormEvent<HTMLButtonElement>) {
     setDrawerOpen(false)
-    onClose(event)
+
+    if (typeof onClose === 'function') {
```

Comment:
👍 don't know if we need to check that it's a function because that's implicit with the strong typing around `onClose`, ie if it's defined then it will definitely be a function